src/lib/kernel/LinuxProcess.cc

```diff
@@ -160,9 +164,23 @@ void LinuxProcess::createStack(char** processImage) {
 
   // ELF auxillary vector, keys defined in `uapi/linux/auxvec.h`
   // TODO: populate remaining auxillary vector entries
-  initialStackFrame.push_back(6);  // AT_PAGESZ
+  initialStackFrame.push_back(auxVec::AT_PHDR);  // AT_PHDR
+  initialStackFrame.push_back(progHeaderTableAddress_);
+
+  initialStackFrame.push_back(auxVec::AT_PHENT);  // AT_PHENT
+  initialStackFrame.push_back(progHeaderEntSize_);
+
+  initialStackFrame.push_back(auxVec::AT_PHNUM);  // AT_PHNUM
+  initialStackFrame.push_back(numProgHeaders_);
+
+  initialStackFrame.push_back(auxVec::AT_PAGESZ);  // AT_PAGESZ
   initialStackFrame.push_back(pageSize_);
-  initialStackFrame.push_back(0);  // null terminator
+
+  initialStackFrame.push_back(auxVec::AT_ENTRY);  // AT_ENTRY
+  initialStackFrame.push_back(entryPoint_);
+
+  initialStackFrame.push_back(auxVec::AT_NULL);  // null terminator
```

Comment:
Do we need the AT_NULL value and the 0?

Reply:
As in figure 22 from [this page](https://refspecs.linuxfoundation.org/ELF/zSeries/lzsabi0_zSeries/x895.html) each auxvec entry is a struct of a_type and a_un so for consistency I added a value here for a_un. In Table 14 however it says that this value is ignored. I think it does work without the 0 but it does keep inline with the struct. Tell me what you think is best

Reply:
Here is a reduced printout of memory at the SP

0x3ffffffb00: 0x0000000000000001 = argc 0x0000003ffffffd21 = argv[0] (points to string containing full path to program) 
0x3ffffffb10: 0x0000000000000000  = NULL 0x0000003ffffffd51 = envp[0]
...
0x3ffffffbd0: 0x0000003fffffff64 0x0000003fffffff9a
0x3ffffffbe0: 0x0000003fffffffb5 0x0000000000000000 = envp NULL
0x3ffffffbf0: 0x0000000000000021 0x0000003ff7ffe000 = auxvec[0]
...
0x3ffffffc10: 0x0000000000000006 0x0000000000001000 = AT_PAGESZ, 4096
...
0x3ffffffcf0: 0x000000000000001f 0x0000003fffffffc8
0x3ffffffd00: 0x0000000000000000 0x0000000000000000 = AT_NULL
0x3ffffffd10: 0x34230600805cf300 0xab5c122963e0332f = Garbage

So the current implementation does follow gdb output